test/yargs.js

```diff
@@ -1494,4 +1494,206 @@ describe('yargs context', function () {
       .argv
     context.commands.should.deep.equal([])
   })
+
+  context('custom exit handler', function () {
+    it('delegates to custom exit when printing completion script', function (done) {
+      var completionScript = null
+      var argv = yargs('completion')
+        .completion()
+        .logger({
+          log: function (_completionScript) {
+            completionScript = _completionScript
+          }
+        })
+        .exit(function (code) {
+          code.should.equal(0)
+          completionScript.should.match(/yargs command completion script/)
+          process.nextTick(done)
+        })
+        .argv
+
+      argv._.should.include('completion')
+    })
+
+    it('delegates to custom exit when making completion suggestions', function (done) {
+      var completions = ''
+      var argv = yargs(['--get-yargs-completions'])
+        .completion('completion', function (current, argv) {
+          return new Promise(function (resolve, reject) {
```

Comment:
Perhaps we should add `require('es6-promise').polyfill()` to this file (`test/yargs.js`) as well.